File: Culinario_DB/EFCore/Tables/Recipes/Recipe.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

// ReSharper disable CheckNamespace
namespace Culinario_DB.EFCore.Tables;

[Table("Recipe")]
public class Recipe : ITable
{
    /// <summary>
    /// Идентификатор рецепта
    /// </summary>
    [Key]
    public int Id { get; set; }

    /// <summary>
    /// Название рецепта
    /// </summary>
    [StringLength(100)]
    public required string Name { get; set; }

    /// <summary>
    /// Описание рецепта
    /// </summary>
    [StringLength(500)]
    public required string Description { get; set; }

    /// <summary>
    /// Категория рецепта
    /// </summary>
    public required RecipeCategory Category { get; set; }

    /// <summary>
    /// Изображение рецепта
    /// </summary>
    public RecipeImage? Image { get; set; }

    /// <summary>
    /// Пользователь, добавивший рецепт
    /// </summary>
    [ForeignKey("UserId")]
    public required User UserId { get; set; }

    /// <summary>
    /// Список продуктов для рецепта
    /// </summary>
    public required List<Product> Products { get; set; }

    /// <summary>
    /// Шаги приготовления рецепта
    /// </summary>
    public List<RecipeSteps>? Steps { get; set; }

    /// <summary>
    /// Количество лайков рецепта
    /// </summary>
    [Range(0, int.MaxValue)]
    public int Likes { get; set; }

    /// <summary>
    /// Количество дизлайков рецепта
    /// </summary>
    [Range(0, int.MaxValue)]
    public int Dislikes { get; set; }

    /// <summary>
    /// Количество просмотров рецепта
    /// </summary>
    [Range(0, int.MaxValue)]
    public int Views { get; set; }

    /// <summary>
    /// Комментарии к рецепту
    /// </summary>
    public List<Comment>? Comments { get; set; }
}

File: Culinario/app/src/main/java/com/culinario/mvp/models/recipes/Recipe.kt
package com.culinario.mvp.models

import androidx.annotation.Size
import kotlinx.serialization.Serializable

/**
 * Класс Recipe представляет рецепт с уникальным идентификатором, названием, описанием, категорией, изображением, идентификатором пользователя, списком продуктов, шагами приготовления, количеством лайков, дизлайков, просмотров и комментариев.
 * @param Id Уникальный идентификатор рецепта.
 * @param Name Название рецепта, длина строки не должна превышать 100 символов.
 * @param Description Описание рецепта, длина строки не должна превышать 500 символов.
 * @param Category Категория рецепта.
 * @param Image Изображение рецепта.
 * @param UserId Идентификатор пользователя, создавшего рецепт.
 * @param Products Список продуктов, используемых в рецепте.
 * @param Steps Список шагов приготовления рецепта.
 * @param Likes Количество лайков рецепта.
 * @param Dislikes Количество дизлайков рецепта.
 * @param Views Количество просмотров рецепта.
 * @param Comments Список комментариев к рецепту.
 */
@Serializable
class Recipe(
    override val Id: Int,
    @Size(max = 100) var Name: String,
    @Size(max = 500) var Description: String,
    var Category: RecipeCategory,
    var Image: RecipeImage?,
    var UserId: User,
    var Products: List<Product>,
    var Steps: List<RecipeSteps>?,
    var Likes: UInt,
    var Dislikes: UInt,
    var Views: UInt,
    var Comments: List<Comment>?,
) : IModel